src/tasks/authenticate/taskpasswordauth.cpp

```diff
@@ -0,0 +1,246 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "taskpasswordauth.h"
+#include "hkdf.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "networkrequest.h"
+#include "networkmanager.h"
+
+#include <QCryptographicHash>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QPasswordDigestor>
+#include <QRandomGenerator>
+#include <QUrl>
+#include <QUrlQuery>
+
+namespace {
+Logger logger(LOG_MAIN, "TaskPasswordAuth");
+}
+
+TaskPasswordAuth::TaskPasswordAuth(const QString& username,
+                                   const QString& password)
+    : Task("TaskPasswordAuth"), m_username(username) {
+  MVPN_COUNT_CTOR(TaskPasswordAuth);
+
```

Comment:
In theory this should be a bit more complex than this because we should:
1. check if the user account exists (/v1/account/status + email)
2. if it doesn't exist, we should create it.

Reply:
I feel that if the account doesn't exist then we should return an authentication failure from this task and inform the user about it (for example, it might have been a typo). Going directly to account creation seems inappropriate for a CLI command.

When run from the GUI, however, we could use the `NoSuchAccount` error status and use it to drive a state change to an account creation view, but that's something that could be added in another PR later.